plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -255,15 +255,17 @@ export const createPublishGithubPullRequestAction = ({
           {
             // See the properties of tree items
             // in https://docs.github.com/en/rest/reference/git#trees
-            mode: file.executable ? '100755' : '100644',
+            mode: file.symlink ? '120000' : (file.executable ? '100755' : '100644'),
             // Always use base64 encoding to avoid doubling a binary file in size
```

Comment:
Despite the mentino here about "Always use base64 encoding", I found that submitting a symlink as base64 would result in scrambled text rather than the proper output

https://user-images.githubusercontent.com/14816406/183278303-5ca4f665-85ad-46fa-8932-7ce193f7d0cf.png